chainimport/headers_import_test.go

```diff
@@ -3020,6 +3020,619 @@ func TestImportAndTargetSourcesHeadersVerificationConstraint(t *testing.T) {
+				bFile, c1, err := setupFileWithHdrs(hT, true)
+				if err != nil {
+					return Prep{
+						cleanup: c1,
+						err:     err,
+					}
+				}
+
+				// Configure import options.
+				opts := &ImportOptions{}
+				bS := opts.createBlockHeaderImportSrc()
+				bS.SetURI(bFile.Name())
+
+				err = bS.Open()
+				c2 := func() {
+					bS.Close()
+					os.Remove(bFile.Name())
+				}
+				if err != nil {
+					return Prep{
+						cleanup: c2,
+						err:     err,
+					}
+				}
+
+				current := uint32(i)
+				end := uint32(len(blockHdrs) - 1)
+				iter := &importSourceHeaderIterator{
+					source:       bS,
+					currentIndex: current,
+					endIndex:     end,
+					batchSize:    128,
+				}
+
+				cleanup := func() {
+					iter.Close()
+					c2()
+				}
+
+				return Prep{
+					hIterator: iter,
+					cleanup:   cleanup,
+				}
+			},
+			verify: func(v Verify) {
+				iter := v.hIterator
+				nBHs := len(blockHdrs)
+				for i := 1; i < nBHs; i++ {
+					h, err := iter.Next()
+					require.NoError(v.tc, err)
+					hE, err := constructBlkHdr(
+						blockHdrs[i], uint32(i),
+					)
+					require.NoError(t, err)
+					require.Equal(v.tc, hE, h)
+				}
+				h, err := iter.Next()
+				require.ErrorIs(v.tc, err, io.EOF)
+				require.Nil(v.tc, h)
+			},
+		},
+		{
+			name:  "IterateOverFilterHeadersSuccessfully",
+			index: 0,
+			hType: headerfs.RegularFilter,
+			prep: func(hT headerfs.HeaderType, i int) Prep {
+				// Create filter headers file.
+				fFile, c1, err := setupFileWithHdrs(
+					hT, true,
+				)
+				if err != nil {
+					return Prep{
+						cleanup: c1,
+						err:     err,
+					}
+				}
+
+				// Configure import options.
+				opts := &ImportOptions{}
+				fS := opts.createFilterHeaderImportSrc()
+				fS.SetURI(fFile.Name())
+
+				err = fS.Open()
+				c2 := func() {
+					fS.Close()
+					os.Remove(fFile.Name())
+				}
+				if err != nil {
+					return Prep{
+						cleanup: c2,
+						err:     err,
+					}
+				}
+
+				current := uint32(i)
+				end := uint32(len(filterHdrs) - 1)
+				iter := &importSourceHeaderIterator{
+					source:       fS,
+					currentIndex: current,
+					endIndex:     end,
+					batchSize:    128,
+				}
+
+				cleanup := func() {
+					iter.Close()
+					c2()
+				}
+
+				return Prep{
+					hIterator: iter,
+					cleanup:   cleanup,
+				}
+			},
+			verify: func(v Verify) {
+				iter := v.hIterator
+				nFHs := len(filterHdrs)
+				for i := 1; i < nFHs; i++ {
+					h, err := iter.Next()
+					require.NoError(v.tc, err)
+					hE, err := constructFilterHdr(
+						filterHdrs[i], uint32(i),
+					)
+					require.NoError(t, err)
+					require.Equal(v.tc, hE, h)
+				}
+				h, err := iter.Next()
+				require.ErrorIs(v.tc, err, io.EOF)
+				require.Nil(v.tc, h)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			prep := tc.prep(tc.hType, tc.index)
+			t.Cleanup(prep.cleanup)
+			require.NoError(t, prep.err)
+
+			header, err := prep.hIterator.Next()
```

Comment:
Is there a particular reason that these all use the `Next()` method explicitly, instead of just using the range syntax?